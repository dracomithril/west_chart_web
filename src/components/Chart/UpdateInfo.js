import React from 'react';
import PropTypes from 'prop-types';
import { Label } from 'react-bootstrap';

const UpdateInfo = ({ last_update, since, until, filtered, total }) => {
  const options = { weekday: 'short', month: '2-digit', day: 'numeric' };
  const last_update_date = last_update ? new Date(last_update).toLocaleString('pl-PL') : 'No data';
  return (
    <div className="update-info">
      <div id="time-frame" className="update-info__time-frame">
        <span>since: </span>
        <Label bsStyle="success">{since !== '' ? new Date(since).toLocaleDateString('pl-PL', options) : 'null'}</Label>
        <span> to </span>
        <Label bsStyle="danger">{until !== '' ? new Date(until).toLocaleDateString('pl-PL', options) : 'null'}</Label>
      </div>
      <span id="updateDate" className="update-info__span">{` Last update: ${last_update_date}`}</span>
      <span className="update-info__span">
        We did get {total} and filtered {filtered}
      </span>
    </div>
  );
};

UpdateInfo.propTypes = {
  last_update: PropTypes.string,
  since: PropTypes.number,
  until: PropTypes.number,
  total: PropTypes.number,
  filtered: PropTypes.number,
};

export default UpdateInfo;
