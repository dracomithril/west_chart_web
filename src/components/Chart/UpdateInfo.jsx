// @flow
import React from 'react';
import { getFormattedDate } from '../../utils/utils';

type Props = {
  since?: number,
  until?: number,
  lastUpdateDateString?: string,
  allCount?: number,
  viewedCount?: number,
};

const UpdateInfo = ({
  since, until, lastUpdateDateString, allCount, viewedCount,
}: Props) => (
  <div className="update-info">
    {(since && until) && (
    <div id="time-frame" className="update-info__time-frame">
      {'since: '}
      <span style={{ color: 'blue' }}>
        { getFormattedDate(since, 'ddd, DD/MM') || 'null'}
      </span>
      {' to '}
      <span style={{ color: 'red' }}>
        { getFormattedDate(until, 'ddd, DD/MM') || 'null'}
      </span>
    </div>
    )}
    {lastUpdateDateString && (
    <span id="updateDate" className="update-info__span">
      {` Last update: ${lastUpdateDateString}`}
    </span>
    )}
    <span className="update-info__span">
      We did get
      {' '}
      {allCount}
      {' '}
and filtered
      {' '}
      {viewedCount}
    </span>
  </div>
);

UpdateInfo.defaultProps = {
  since: 0,
  until: 0,
  lastUpdateDateString: '',
  allCount: 0,
  viewedCount: 0,
};

export default UpdateInfo;
