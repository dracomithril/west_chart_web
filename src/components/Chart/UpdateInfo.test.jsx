import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import UpdateInfo from './UpdateInfo';

Enzyme.configure({ adapter: new Adapter() });
describe('<UpdateInfo/>', () => {
  it('renders without crashing UpdateInfo', () => {
    const wrapper = shallow(
      <UpdateInfo />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('renders users', () => {
    const lastUpdateDateString = '2017-03-16T13:46:00.570Z';
    const since = new Date('2017-03-11T13:46:00.570Z').valueOf();
    const until = new Date(lastUpdateDateString).valueOf();
    const wrapper = shallow(
      <UpdateInfo
        since={since}
        lastUpdateDateString={lastUpdateDateString}
        until={until}
        allCount={50}
        viewedCount={4}
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
