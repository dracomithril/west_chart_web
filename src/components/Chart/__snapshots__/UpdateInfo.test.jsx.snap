// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateInfo/> renders users 1`] = `
<div
  className="update-info"
>
  <div
    className="update-info__time-frame"
    id="time-frame"
  >
    since: 
    <span
      style={
        Object {
          "color": "blue",
        }
      }
    >
      Sat, 11/03
    </span>
     to 
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      Thu, 16/03
    </span>
  </div>
  <span
    className="update-info__span"
    id="updateDate"
  >
     Last update: 2017-03-16T13:46:00.570Z
  </span>
  <span
    className="update-info__span"
  >
    We did get
     
    50
     
    and filtered
     
    4
  </span>
</div>
`;

exports[`<UpdateInfo/> renders without crashing UpdateInfo 1`] = `
<div
  className="update-info"
>
  0
  <span
    className="update-info__span"
  >
    We did get
     
    0
     
    and filtered
     
    0
  </span>
</div>
`;
